src/include/execution/tplclass.h

```diff
@@ -0,0 +1,422 @@
+
+                // Dump Bytecode
+                if (kPrintTbc) {
+                    bytecode_module->PrettyPrint(&std::cout);
+                }
+                modules_[num_modules_] = std::make_unique<vm::Module>(std::move(bytecode_module));
+                name_id_[filename] = num_modules_;
+                module_id = num_modules_;
+                num_modules_++;
+            } else {
+                module_id = itr->second;
+            }
+
+/*
+            //
+            // Interpret
+            //
+
+            {
+                util::ScopedTimer<std::milli> timer(&interp_exec_ms);
+
+                if (kIsSQL) {
+                    std::function<i64(exec::ExecutionContext *)> main;
+                    if (!module->GetFunction("main", vm::ExecutionMode::Interpret, &main)) {
+                        EXECUTION_LOG_ERROR(
+                                "Missing 'main' entry function with signature "
+                                "(*ExecutionContext)->int64");
+                        return;
+                    }
+                    auto memory = std::make_unique<sql::MemoryPool>(nullptr);
+                    exec_ctx.SetMemoryPool(std::move(memory));
+                    EXECUTION_LOG_INFO("VM main() returned: {}", main(&exec_ctx));
+                } else {
+                    std::function<i64()> main;
+                    if (!module->GetFunction("main", vm::ExecutionMode::Interpret, &main)) {
+                        EXECUTION_LOG_ERROR("Missing 'main' entry function with signature ()->int64");
+                        return;
+                    }
+                    EXECUTION_LOG_INFO("VM main() returned: {}", main());
+                }
+            }
+
+            //
+            // Adaptive
+            //
+
+            {
+                util::ScopedTimer<std::milli> timer(&adaptive_exec_ms);
+
+                if (kIsSQL) {
+                    std::function<i64(exec::ExecutionContext *)> main;
+                    if (!module->GetFunction("main", vm::ExecutionMode::Adaptive, &main)) {
+                        EXECUTION_LOG_ERROR(
+                                "Missing 'main' entry function with signature "
+                                "(*ExecutionContext)->int64");
+                        return;
+                    }
+                    auto memory = std::make_unique<sql::MemoryPool>(nullptr);
+                    exec_ctx.SetMemoryPool(std::move(memory));
+                    EXECUTION_LOG_INFO("ADAPTIVE main() returned: {}", main(&exec_ctx));
+                } else {
+                    std::function<i64()> main;
+                    if (!module->GetFunction("main", vm::ExecutionMode::Adaptive, &main)) {
+                        EXECUTION_LOG_ERROR("Missing 'main' entry function with signature ()->int64");
+                        return;
+                    }
+                    EXECUTION_LOG_INFO("ADAPTIVE main() returned: {}", main());
+                }
+            }
```

Comment:
Add another flag as a member variable to select which mode to run, instead of leaving commented code in the file.